components/concept/sync/src/main/java/mozilla/components/concept/sync/OAuthAccount.kt

```diff
@@ -33,20 +33,155 @@ data class AuthFlowUrl(val state: String, val url: String)
  */
 @SuppressWarnings("TooManyFunctions")
 interface OAuthAccount : AutoCloseable {
+
+    /**
+     * Constructs a URL used to begin the OAuth flow for the requested scopes and keys.
+     *
+     * @param scopes List of OAuth scopes for which the client wants access
+     * @return Deferred AuthFlowUrl that resolves to the flow URL when complete
+     */
     fun beginOAuthFlowAsync(scopes: Set<String>): Deferred<AuthFlowUrl?>
+
+    /**
+     * Constructs a URL used to begin the pairing flow for the requested scopes and pairingUrl.
+     *
+     * @param pairingUrl URL string for pairing
+     * @param scopes List of OAuth scopes for which the client wants access
+     * @return Deferred AuthFlowUrl Optional that resolves to the flow URL when complete
+     */
     fun beginPairingFlowAsync(pairingUrl: String, scopes: Set<String>): Deferred<AuthFlowUrl?>
+
+    /**
+     * Returns current FxA Device ID for an authenticated account.
+     *
+     * @return Current device's FxA ID, if available. `null` otherwise.
+     */
     fun getCurrentDeviceId(): String?
+
+    /**
+     * Returns session token for an authenticated account.
+     *
+     * @return Current account's session token, if available. `null` otherwise.
+     */
     fun getSessionToken(): String?
+
+    /**
+     * Returns an authorized auth code for the passed scopes
+     *
+     * @param clientId the client id string
+     * @param scopes the list of scopes to request access to
+     * @param state the state token string
+     * @param accessType the accessType method to be used by the returned code, "offline" or "online"
+     * @return the authorized auth code string
+     */
+    fun authorizeOAuthCode(
+        clientId: String,
+        scopes: Array<String>,
+        state: String,
+        accessType: String = "online"
+    ): String?
+
+    /**
+     * Fetches the profile object for the current client either from the existing cached account,
+     * or from the server (requires the client to have access to the profile scope).
+     *
+     * @param ignoreCache Fetch the profile information directly from the server
+     * @return Profile (optional, if successfully retrieved) representing the user's basic profile info
+     */
     fun getProfileAsync(ignoreCache: Boolean): Deferred<Profile?>
+
+    /**
+     * Convenience method to fetch the profile from a cached account by default, but fall back
+     * to retrieval from the server.
+     *
+     * @return Profile (optional, if successfully retrieved) representing the user's basic profile info
+     */
     fun getProfileAsync(): Deferred<Profile?>
+
+    /**
+     * Authenticates the current account using the code and state parameters fetched from the
```

Comment:
This doc is incorrect for the most common case - WebChannels. Let's make it more generic:

`Authenticates the current account using the [code] and [state] parameters obtained via the OAuth flow initiated by [beginOAuthFlowAsync].`

Reply:
If you can, please update the original kdoc in  `FirefoxAccount` as well.